ingester/ingester_lifecycle_test.go

```diff
@@ -0,0 +1,99 @@
+package ingester
+
+import (
+	"reflect"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/log"
+	"github.com/weaveworks/cortex/ring"
+)
+
+func defaultIngesterTestConfig() Config {
+	consul := ring.NewMockConsulClient()
+	return Config{
+		ringConfig: ring.Config{
+			ConsulConfig: ring.ConsulConfig{
+				Mock: consul,
+			},
+		},
+
+		NumTokens:       1,
+		HeartbeatPeriod: 5 * time.Second,
+		ListenPort:      func(i int) *int { return &i }(0),
+		Addr:            "localhost",
+		Hostname:        "localhost",
+
+		FlushCheckPeriod: 99999 * time.Hour,
+		MaxChunkIdle:     99999 * time.Hour,
+	}
+}
+
+func TestIngesterRestart(t *testing.T) {
+	config := defaultIngesterTestConfig()
+	config.skipUnregister = true
+
+	{
+		ingester, err := New(config, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
```

Comment:
Thanks! If you're really keen, you can also import `assert` from testify and use its `Equal` function to replace some of the `DeepEqual` business.